client/components/search/index.jsx

```diff
@@ -255,13 +262,13 @@ module.exports = React.createClass( {
 						? this._keyListener.bind( this, 'openSearch' )
 						: null
 					}
-					aria-controls={ 'search-component-' + this.id }
+					aria-controls={ 'search-component-' + this.state.instanceId }
```

Comment:
> It could be worth whipping up a /design SSR attempt now, just to see what happens with Search reconc.

I suggest tackling this later if it becomes a problem.